src/main/java/org/truffleruby/core/VMPrimitiveNodes.java

```diff
@@ -343,34 +341,18 @@ public Object get(DynamicObject key) {
     @Primitive(name = "vm_get_config_section", needsSelf = false)
     public abstract static class VMGetConfigSectionPrimitiveNode extends PrimitiveArrayArgumentsNode {
 
-        @Child private org.truffleruby.core.string.StringNodes.MakeStringNode makeStringNode = org.truffleruby.core.string.StringNodes.MakeStringNode.create();
+        @Child private StringNodes.MakeStringNode makeStringNode = StringNodes.MakeStringNode.create();
+        @Child private YieldNode yieldNode = new YieldNode();
 
         @TruffleBoundary
-        @Specialization(guards = "isRubyString(section)")
-        public DynamicObject getSection(DynamicObject section) {
-            final List<DynamicObject> sectionKeyValues = new ArrayList<>();
-
+        @Specialization(guards = { "isRubyString(section)", "isRubyProc(block)" })
+        public DynamicObject getSection(DynamicObject section, DynamicObject block) {
             for (Entry<String, Object> entry : getContext().getRubiniusConfiguration().getSection(StringOperations.getString(section))) {
-                Object value = entry.getValue();
-                final String stringValue;
-                if (RubyGuards.isRubyBignum(value)) {
-                    stringValue = Layouts.BIGNUM.getValue((DynamicObject) value).toString();
-                } else if (RubyGuards.isRubyString(value)) {
-                    stringValue = StringOperations.getString((DynamicObject) value);
-                } else if (value instanceof Boolean || value instanceof Integer || value instanceof Long) {
-                    stringValue = value.toString();
-                } else {
-                    throw new UnsupportedOperationException(value.getClass().toString());
-                }
-
-                Object[] objects = new Object[] {
-                        makeStringNode.executeMake(entry.getKey(), UTF8Encoding.INSTANCE, CodeRange.CR_7BIT),
-                        makeStringNode.executeMake(stringValue, UTF8Encoding.INSTANCE, CodeRange.CR_7BIT) };
-                sectionKeyValues.add(createArray(objects, objects.length));
+                final DynamicObject key = makeStringNode.executeMake(entry.getKey(), UTF8Encoding.INSTANCE, CodeRange.CR_7BIT);
```

Comment:
I don't know the answer to this, but should the encoding be `Encoding.default_internal` instead of the hard-coded UTF8? If so, you'd have to update the code range as well, since 7-bit is only useful for ASCII-compatible encodings.

Reply:
It's all ASCII-only strings, see `LinuxRubiniusConfiguration` for example.
We could tag them with a US-ASCII encoding, it doesn't matter much I think.
Using UTF-8 makes it coherent with the encoding of the String literals in Ruby used to access it.